### PR TITLE
Boundaries list

### PR DESCRIPTION
**Pull request by zonca**
Reimplemented #271 with `np.can_cast` and `np.isscalar`

### CODE REVIEW

**Comment by zonca** [comment]
@lpsinger would you take a look at this? I'd like to release a new version of healpy today

**Comment by lpsinger** [comment]
Sorry, no, I cannot review this today. I could look at it tomorrow.

On Sep 17, 2015, at 12:05 PM, Andrea Zonca <notifications@github.com<mailto:notifications@github.com>> wrote:


@lpsinger<https://github.com/lpsinger> would you take a look at this? I'd like to release a new version of healpy today

**Comment by zonca** [comment]
@lpsinger ok! thanks!

File: healpy/src/_query_disc.pyx [hunk 1]
```diff
@@ -277,14 +277,18 @@ def boundaries(nside, pix, step=1, nest=False):
...
+        if not np.can_cast(type(pix), np.int):
+            raise ValueError('Array of pixels has to be integers')
         return _boundaries_single(nside, pix, step=step, nest=nest)
-    if type(pix) is np.ndarray:
-        if not issubclass(pix.dtype.type, np.integer):
+    else:
+        pix = np.asarray(pix)
+        if np.can_cast(pix.dtype, np.int):
+            if pix.ndim!=1:
+                raise ValueError('Array has to one dimensional')
```
Comment (lpsinger) [inline healpy/src/_query_disc.pyx]: Array has **to be** one dimensional

**Pull request merged**